src/pages/RFOX/hooks/useRfoxClaimActionSubscriber.tsx

```diff
@@ -0,0 +1,95 @@
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+// Note: the is a slight misnomer here compared to what we conventionally call a subscriber in action center (a poller)
+// However, this *is* reactive on useGetUnstakingRequestsQuery(), with inner queries which do get invalidated, so it is a subscriber in a way, just different
+
+export const useRfoxClaimActionSubscriber = () => {
+  const translate = useTranslate()
+  const dispatch = useAppDispatch()
+  const assets = useAppSelector(selectAssets)
+
+  const allUnstakingRequests = useGetUnstakingRequestsQuery()
+
+  const pendingRfoxClaimActions = useAppSelector(selectPendingRfoxClaimActions)
+  const actionIds = useAppSelector(actionSlice.selectors.selectActionIds)
+
+  useEffect(() => {
+    if (!pendingRfoxClaimActions.length) return
+    const now = Date.now()
+
+    pendingRfoxClaimActions.forEach(action => {
+      if (!action.rfoxClaimActionMetadata.txHash) return
+      const asset = assets[action.rfoxClaimActionMetadata.request.stakingAssetId]
+      if (!asset) return
+
+      const amountCryptoPrecision = fromBaseUnit(
+        action.rfoxClaimActionMetadata.request.amountCryptoBaseUnit,
+        asset.precision,
+      )
+
+      dispatch(
+        actionSlice.actions.upsertAction({
+          id: action.id,
+          status: ActionStatus.Claimed,
+          type: ActionType.RfoxClaim,
+          createdAt: action.createdAt,
+          updatedAt: now,
+          rfoxClaimActionMetadata: {
+            ...action.rfoxClaimActionMetadata,
+            message: `Your claim of ${Number(amountCryptoPrecision).toFixed(2)} ${
+              asset.symbol
+            } is complete`,
+          },
+        }),
+      )
+    })
+    // We definitely don't want to react on assets here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pendingRfoxClaimActions, dispatch])
+
+  useEffect(() => {
+    if (!allUnstakingRequests.isSuccess) return
+    const now = Date.now()
+
+    allUnstakingRequests.data.all.forEach(request => {
+      const cooldownExpiryMs = Number(request.cooldownExpiry) * 1000
+
+      if (now >= cooldownExpiryMs) {
+        const asset = assets[request.stakingAssetId]
+        if (!asset) return
+
+        const amountCryptoPrecision = fromBaseUnit(request.amountCryptoBaseUnit, asset.precision)
+
+        dispatch(
+          actionSlice.actions.upsertAction({
+            id: request.id,
+            status: ActionStatus.ClaimAvailable,
+            type: ActionType.RfoxClaim,
+            createdAt: cooldownExpiryMs,
+            updatedAt: now,
+            rfoxClaimActionMetadata: {
+              request,
+              message: translate('notificationCenter.rfox.unstakeReady', {
+                amount: bn(amountCryptoPrecision).toFixed(6),
+                symbol: asset.symbol,
+              }),
+            },
+          }),
+        )
+      }
+    })
+    // We definitely don't want to react on assets here
```

Comment:
Are we 100% sure assets are loaded at this point? feels weird

Reply:
Two things here:
1. stale `assets` shouldn't be an issue: just because this doesn't react on assets doesn't mean that this will use a stale `assets` reference, it will always use the newest one (this isn't a closure on deps), just that the effect won't re-run on `assets` change, which is definitely what we want here as this would cause a virtually infinite loop.

2. re: the actual assets we need (FOX / WETH/FOX LP) are amongst the first populated assets (initial assets upsertion) and are also populated from your portfolio. Dealing with RFOX claims also implies having ETH / WETH/FOX, as that's inherent to the feature as a whole.
In other words, if we ever hit this effect, it means we already have those assets in store!